scripts/builder/checker_builder/config.py

```diff
@@ -114,10 +143,37 @@ class Constants:
 
 @dataclass(frozen=True)
 class CheckerConfig:
+    collateral_type: CollateralType
     tokens: Tokens
     constants: Constants
     drift_derivative_curve: Union[BangBang, Continuous]
 
+    def __post_init__(self) -> None:
+        # Validation logic
+        wtez = self.tokens.issued.wtez
+        wctez = self.tokens.issued.wctez
+        collateral = self.tokens.in_use.collateral
+        cfmm_token = self.tokens.in_use.cfmm_token
+
+        if self.collateral_type == CollateralType.TEZ:
+            if not (collateral.token_id == wtez.token_id) and (
+                collateral.decimal_digits == wtez.decimal_digits
+            ):
```

Comment:
Hmmm, shouldn't the not scope over the whole condition? I mean like this:

```suggestion
            if not ((collateral.token_id == wtez.token_id) and (
                collateral.decimal_digits == wtez.decimal_digits
            )):
```

Reply:
Ah yes it absolutely should. Good catch!